src/backend/parser/gram.y

```diff
@@ -14539,10 +14539,20 @@ a_expr:		c_expr									{ $$ = $1; }
 				}
 			| a_expr AT TIME ZONE a_expr			%prec AT
 				{
+					if(sql_dialect == SQL_DIALECT_TSQL)
```

Comment:
Why can't we add this support in ANTLR?

Reply:
I tried doing the change but it seem a little more then expected as per the work I did so tried using engine code .